File: 28-implement-strstr/28.implement-str-str.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Implement strStr()
 */

// @lc code=start
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle == null || needle.length() == 0) {
            return 0;
        }
        int m = haystack.length(), n = needle.length();
        for(int i = 0; ; i++) {
            for(int j = 0;;j++) {
                if(j == needle.length()) {
                    return i;
                }

                if(i + j == m) {
                    return -1;
                }
                if(haystack.charAt(i+j) != needle.charAt(j)) {
                    break;
                }

            }

        }
    }
}
// @lc code=end

File: algorithm/28.implement-strstr/28-implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if(!needle) {
        return 0;
    }
    for(let i = 0; i < haystack.length; i++) {
        let j = 0;
        for(; j < needle.length; j++) {
            if(i+j===haystack.length)  {
                return -1;
            }
            if(haystack[i+j] !== needle[j]) {
                break;
            }
        }
        if(j===needle.length) {
            return i;
        }
    }
    return -1;
};

